pandas/_libs/tslibs/conversion.pyx

```diff
@@ -636,34 +636,41 @@ cdef inline int64_t[:] _tz_convert_dst(int64_t[:] values, tzinfo tz,
     """
     cdef:
         Py_ssize_t n = len(values)
-        Py_ssize_t i, pos
+        Py_ssize_t i
+        int64_t[:] pos
```

Comment:
Design question here: why do we need to allocate an array for `pos` ?  It seems like we're trading off memory for performance, are we not?

Reply:
We're now having `np.searchsorted()` compute insertion indices for every value in a single call, rather than making `n` `np.searchsorted()` calls. This requires `n` memory and allows `np.searchsorted()` to re-use partial solution information.

And yes... though this would be the 4th such array in this function so it's not changing the broad characteristics here.

Reply:
> though this would be the 4th such array in this function so it's not changing the broad characteristics here.

That is a fair point.